SECURITY.md

```diff
@@ -0,0 +1,79 @@
+how vulnerabilities should be reported.
+
+
+## Vulnerability Definition
+
+WebGPU introduces a different threat model than is sometimes applied to
+GPU-related software. Unlike typical gaming or high-performance computing
+applications, where the software accessing GPU APIs is proprietary or
+obtained from a trusted developer, WebGPU makes GPU APIs available to
+arbitrary web applications. In the threat model of the web, malicious
+content should not be able to use the GPU APIs to access data or interfaces
+outside the intended scope for interaction with web content.
+
+While the WGPU maintainers have discretion in assigning a severity to
+individual vulnerabilities, it is generally considered a security vulnerability
+in WGPU if Javascript or WebAssembly code running with privileges of ordinary
+web content is able to:
+
+- Access data associated with native applications other than the user agent,
+  or associated with other web origins.
+- Escape the applicable sandbox and run arbitrary code or call arbitrary system
+  APIs on the user agent host.
+- Consume system resources to the point that it is difficult to recover
+  (e.g. by closing the web page).
+
+The WGPU Rust API offers some functionality, both supported and experimental,
+that is not part of the WebGPU standard and is not made available in Javascript
+environments using WGPU. Associated vulnerabilities may be assigned lower
+severity than vulnerabilities that apply to a Javascript WebGPU implementation
+based on WGPU.
+
+
+## Supported Versions
+
+The WGPU project maintains security support for serious vulnerabilities in the
+[most recent major release](https://github.com/gfx-rs/wgpu/releases). Fixes for
+security vulnerabilities found shortly after the initial release of a major
+version may also be provided for the previous major release.
+
+Mozilla provides security support for versions of WGPU used in [current
+versions of Firefox](https://whattrainisitnow.com/).
+
+The version of WGPU that is active can be found in the Firefox repositories:
+
+- [release](https://github.com/mozilla-firefox/firefox/blob/release/gfx/wgpu_bindings/Cargo.toml),
+- [beta](https://github.com/mozilla-firefox/firefox/blob/beta/gfx/wgpu_bindings/Cargo.toml), and
+- [nightly](https://github.com/mozilla-firefox/firefox/blob/main/gfx/wgpu_bindings/Cargo.toml),
+
+We welcome reports of security vulnerabilities in any of these released
+versions or in the latest code on the `trunk` branch.
+
+
+## Reporting a Vulnerability
+
+Although not all vulnerabilities in WGPU will affect Firefox, Mozilla accepts
+all vulnerability reports for WGPU and directs them appropriately. Additionally,
+Mozilla serves as the CVE numbering authority for the WGPU project.
+
+To report a security problem with WGPU, create a bug in Mozilla's Bugzilla
+instance in the [Core ::
+Graphics :: WebGPU](https://bugzilla.mozilla.org/enter_bug.cgi?product=Core&component=Graphics%3A+WebGPU)
```

Comment:
suggestion: This is a good link, but per the guidance below, I think it'd be even better to bake the security-relevant groups (`core-security`, `gfx-core-security`) into the link, too, so they're ticked by default, i.e.:

https://bugzilla.mozilla.org/enter_bug.cgi?product=Core&component=Graphics%3A+WebGPU&groups=core-security&groups=gfx-core-security

One less potential ~~paper cut~~ big problem to run into. 🙂

Reply:
I don't know what we might do based on the URL behind the scenes, but the checkbox for the graphics security bug group only shows up if you're in the group. I assume the desire is to have a single ingress point for reports, or maybe that's just a default behavior of groups in bugzilla without any intention for the graphics security group in particular.

Reply:
Just tried running through my proposed link through a private window. Looks like it doesn't check the box like it does when using an employee account. Nuts! It doesn't seem to hurt anything (the end result was [bug 1979455](https://bugzilla.mozilla.org/show_bug.cgi?id=1979455)), but it also doesn't actually solve anything for the audience I was hoping for, either. 🤔

I guess I'd suggest using the URL anyway, since at least it'd save a step for folks who already have group membership? Not firmly set on that, though. 🤷🏻‍♂️

Reply:
I think I meant to reply to this before, but didn't. Having them both specified in the URL seems like it results in correct labeling of none of the bugs -- if the user doesn't have permission, then the bug is labeled core-security only; if the user does have permission, the bug gets both labels and still needs to be removed from core-security. Although I guess we can have a bot do that. Okay, I'll add gfx-sec to the URL.